### PR TITLE
Loadout params

### PR DESCRIPTION
**Pull request by bhollis**
* This switches Loadout Optimizer over to using the LoadoutParameters settings object. This keeps settings together and properly persists Ignored stats.
* LoadoutParameters from saved loadouts are dumped into LO state when you reload the loadout via the "Open in Optimizer" button, allowing you to quickly find better armor for the same mods & stats.
* When saving a loadout from LO we now persist all the parameters, including pinned exotic.
* You can now share an account-independent link to your LO settings with other people. This allows linking directly to great mod combos with stat priorities and exotic as well.

This isn't the end of these changes, just the beginning - I still need to change a lot of the LO state to better align with LoadoutParameters and enhance the experience of re-editing a loadout.

### CODE REVIEW

File: src/app/loadout-builder/loadout-params.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+export function migrateLoadoutParametersFromSettings(settings: Settings): Settings {
+  // Only migrate if there aren't already settings
+  if (_.isEmpty(settings.loParameters)) {
+    return {
+      ...settings,
+      loParameters: {
+        statConstraints: settings.loStatSortOrder.map((statHash) => ({
+          statHash,
+        })),
+        upgradeSpendTier: settings.loUpgradeSpendTier,
```
Comment (ryan-rushton) [inline src/app/loadout-builder/loadout-params.ts:73]: Remember when I removed that enum value and have had to put the guard in, in the event that someone had that version somehow. That could potentially be migrated to here.

Comment (bhollis) [reply src/app/loadout-builder/loadout-params.ts]: Got it. I've updated it and we can remove it from the code after a bit

**Review by ryan-rushton — APPROVED** [review]
Looks good, had a bit of a click around and everything is working nicely from what I can see. 

My only question and its a bit of a strange one but is there any chance we can hit the url size limit? A quick google tells me that is 2048 chars.

File: src/app/loadout-builder/loadout-params.ts (hunk 1 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/loadout-params.ts]: Sounds great

**Comment by bhollis** [comment]
I think we could! We'll just have to see.

**Pull request merged**